utils/build/docker/golang/parametric/datadog.go

```diff
@@ -42,8 +42,13 @@ func (s *apmClientServer) startSpanHandler(w http.ResponseWriter, r *http.Reques
 func (s *apmClientServer) StartSpan(ctx context.Context, args *StartSpanArgs) (ddtrace.Span, error) {
 	var opts []tracer.StartSpanOption
 	if p := args.ParentId; p > 0 {
-		parent := s.spans[p]
-		opts = append(opts, tracer.ChildOf(parent.Context()))
+		if span, ok := s.spans[p]; ok {
+			opts = append(opts, tracer.ChildOf(span.Context()))
+		} else if spanContext, ok := s.spanContexts[p]; ok {
+			opts = append(opts, tracer.ChildOf(spanContext))
```

Comment:
Why are we storing/updating spanContexts on the server?

Reply:
The extract/injector headers endpoints are now configured to create a SpanContext. This SpanContext will be stored in a list on the sever (similar to how spans are currently stored). This will allow us to remove header extraction logic from start span. 

All tracing libraries (except php) expose a SpanContext  in their public API (or an equivalent object like TraceDigest). This change will allow use to introduce parametric endpoints that test this interface.